### PR TITLE
[FRONT-1495] Sponsorships page pagination

### PR DESCRIPTION
**Pull request by pmichalowski**

### CODE REVIEW

File: src/network/hooks/useSponsorshipsList.ts [hunk 1]
```diff
@@ -13,19 +13,25 @@ import { getAllSponsorships, getSponsorshipsByCreator } from '~/getters'
 export const useAllSponsorshipsQuery = (
     pageSize = 10,
     searchQuery?: string,
-): UseInfiniteQueryResult<SponsorshipElement[]> => {
+): UseInfiniteQueryResult<{ pageParam: number; elements: SponsorshipElement[] }> => {
```
Comment (mondoreale) [inline src/network/hooks/useSponsorshipsList.ts:16]: `pageParam` → `page`?

File: src/getters/index.ts [hunk 2]
```diff
@@ -240,7 +240,7 @@ export function getGraphProjectWithParsedMetadata<T extends { metadata: string }
 
 export async function getAllSponsorships({
     first,
-    skip,
+    skip = 0,
```
Comment (mondoreale) [inline src/getters/index.ts:243]: Won't `undefined` do the same?

File: src/network/pages/SponsorshipsPage.tsx [hunk 3]
```diff
@@ -63,7 +63,7 @@ export const SponsorshipsPage = () => {
             : mySponsorshipsQuery
 
     const sponsorships: SponsorshipElement[] =
-        (sponsorshipsQuery?.data?.pages.flat() as SponsorshipElement[]) || []
+        sponsorshipsQuery?.data?.pages.map((page) => page.elements).flat() || []
```
Comment (mondoreale) [inline src/network/pages/SponsorshipsPage.tsx:66]: `sponsorshipsQuery` isn't nullish.

```suggestion
        sponsorshipsQuery.data?.pages.map((page) => page.elements).flat() || []
```

File: src/network/pages/SponsorshipsPage.tsx [hunk 4]
```diff
@@ -230,6 +230,14 @@ export const SponsorshipsPage = () => {
                         }
                     />
                 </SponsorshipsTableWrap>
+                {sponsorshipsQuery.hasNextPage && (
+                    <LoadMoreButton
+                        onClick={() => sponsorshipsQuery.fetchNextPage()}
```
Comment (mondoreale) [inline src/network/pages/SponsorshipsPage.tsx:235]: `sponsorshipsQuery.fetchNextPage()` returns a promise. What do you think about try-catching it and if it throws showing an error toast?

File: src/network/pages/SponsorshipsPage.tsx [hunk 5]
```diff
@@ -230,6 +230,14 @@ export const SponsorshipsPage = () => {
...
+                        kind={'primary2'}
```
Comment (mondoreale) [inline src/network/pages/SponsorshipsPage.tsx:236]:
```suggestion
                        kind="primary2"
```

File: src/getters/index.ts (hunk 2 above)
Comment (pmichalowski) [reply src/getters/index.ts:243]: true

File: src/network/hooks/useSponsorshipsList.ts (hunk 1 above)
Comment (pmichalowski) [reply src/network/hooks/useSponsorshipsList.ts:16]: yeah - kinda got too inspired by the tansack query's property names :D I'll change it to what it actually is - `skippedElements`

Comment (mondoreale) [reply src/network/hooks/useSponsorshipsList.ts:16]: It'd a number of skipped elements. 😃

**Pull request merged**